obspy/io/sac/sactrace.py

```diff
@@ -1092,7 +1092,7 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                 val = _ut._clean_str(val, strip_whitespace=False)
                 if val.startswith(native_str('-12345')):
                     val = HD.SNULL
-                hs[i] = val
+                hs[i] = val.encode('ASCII', 'replace')
```

Comment:
OK, upon further investigation, this works because SAC I/O is not very good at defining boundaries and does encode/decode a bit too much.